src/js/overlay.js

```diff
@@ -101,319 +102,328 @@ var Overlay = function(id, opts) {
 	}
 	this.context = this.opts.arrow ? oLayers.getLayerContext(this.opts.arrow.target) : oLayers.getLayerContext(this.opts.trigger);
 
+	this.delegates = {
+		doc: new Delegate(),
+		wrap: new Delegate(),
+		context: new Delegate()
+	};
+
 	// Add this overlay to the overlays hashmap
 	overlays[id] = this;
 };
 
-Overlay.prototype = {
-
-	open: function() {
-		if (!this.content) {
-			var self = this;
-			this.loadContent(function(html) {
-				self.opts.html = html;
-				if (!self.opts.html) {
-					throw new Error('"o-overlay error": Content for the overlay needs to be set via the "html" or the "src" option.');
-				}
-				self.render();
+Overlay.prototype.open = function() {
+	if (!this.content) {
+		var overlay = this;
+		this.loadContent(function(html) {
+			overlay.opts.html = html;
+			if (!overlay.opts.html) {
+				throw new Error('"o-overlay error": Content for the overlay needs to be set via the "html" or the "src" option.');
+			}
+			overlay.render();
+		});
+	} else {
+		this.show();
+	}
+};
+
+Overlay.prototype.loadContent = function(callback) {
+	if (!this.opts.html && this.opts.src) {
+		if (/^(https?\:\/)?\//.test(this.opts.src)) {
+			utils.copyContentFromUrl(this.opts.src, function(html) {
+				callback(html);
 			});
 		} else {
-			this.show();
+			utils.copyContentFromElement(document.querySelector(this.opts.src), function(html) {
+				callback(html);
+			});
 		}
-	},
-
-	loadContent: function(callback) {
-		if (!this.opts.html && this.opts.src) {
-			if (/^(https?\:\/)?\//.test(this.opts.src)) {
-				utils.copyContentFromUrl(this.opts.src, function(html) {
-					callback(html);
-				});
-			} else {
-				utils.copyContentFromElement(document.querySelector(this.opts.src), function(html) {
-					callback(html);
-				});
-			}
-		} else {
-			callback(this.opts.html);
+	} else {
+		callback(this.opts.html);
+	}
+};
+
+Overlay.prototype.render = function() {
+	var wrapperEl = document.createElement('div');
+	wrapperEl.className = 'o-overlay';
+	this.wrapper = wrapperEl;
+
+	if (this.opts.heading) {
+		var heading = document.createElement('header');
+		heading.classList.add('o-overlay__heading');
+
+		if (this.opts.heading.shaded) {
+			heading.classList.add('o-overlay__heading--shaded');
 		}
-	},
 
-	render: function() {
-		var wrapperEl = document.createElement('div');
-		wrapperEl.className = 'o-overlay';
-		this.wrapper = wrapperEl;
+		var button = document.createElement('button');
+		button.className = 'o-overlay__close';
+		var buttonIcon = document.createElement('div');
+		buttonIcon.className = 'o-overlay__close-icon';
+		button.appendChild(buttonIcon);
 
-		if (this.opts.heading) {
-			var heading = document.createElement('header');
-			heading.classList.add('o-overlay__heading');
+		var title = document.createElement('span');
```

Comment:
Because it's just text mainly, the header is divided between the actual heading and the close icon. This can be raised as a separate issue if you think another element is more appropriate